resource_detectors/lib/opentelemetry/resource/detectors/deployment.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module OpenTelemetry
+  module Resource
+    module Detectors
+      # Deployment contains detect class method for determining deployment resource attributes
+      module Deployment
+        extend self
+
+        def detect
+          resource_attributes = {}
+          deployment_environment = rails_deployment_environment || rack_env
+          resource_attributes[::OpenTelemetry::SemanticConventions::Resource::DEPLOYMENT_ENVIRONMENT] = deployment_environment.to_str if deployment_environment
+          OpenTelemetry::SDK::Resources::Resource.create(resource_attributes)
+        end
+
+        private
+
+        def rails_deployment_environment
```

Comment:
For symmetry with `rack_env` 😄 

```suggestion
        def rails_env
```

Reply:
good point :) renamed it to be consistent